rohmu/object_storage/base.py

```diff
@@ -55,6 +55,9 @@ class IterKeyItem(NamedTuple):
 # Argument is the additional number of bytes transferred
 IncrementalProgressCallbackType = Optional[Callable[[int], None]]
 
+# Argument is the transferred object key
+ObjectTransferProgressCallbackType = Optional[Callable[[str], None]]
```

Comment:
nit: could use a `TypeAlias` annotation
```suggestion
from typing import TypeAlias
ObjectTransferProgressCallbackType: TypeAlias = Optional[Callable[[str], None]]
```

Reply:
`TypeAlias` is introduced in Python 3.10+, however we support 3.8+. Probably thats why we haven' t used it.
Added `TypeAlias` import from `typing_extensions`.